### PR TITLE
xtensa-build-zephyr.py: de-hardcode mode=511 in mkdir()

### PR DESCRIPTION
**Pull request by marc-hb**
These hardcoded 511 permissions came with the initial commit 1de3ef367584 ("Rewritten xtensa-build-zephyr.sh to python") without any particular explanation.

On Unix, permissions of new files and directories are a user preference set with the `umask`, different applications are not supposed to create directories differently.

On Windows/NTFS it's not clear what 511 maps to.

Signed-off-by: Marc Herbert <marc.herbert@intel.com>

### CODE REVIEW

**Comment by lgirdwood** [comment]
@aborisovich any insights here ? Could this be we need an ```if WIN32``` for different permissions on Windows ?

**Review by aborisovich — APPROVED** [review]
I'm quite sure that I've tried to preserve permissions from original xtensa-build-zephyr.sh script.
I think it is not needed now context changed very much since then. I think if could be done to have a warranty that
west_top directory name will not be changed externally? (It would broke script execution obviously).
Good to remove.

On Windows it looks like it does nothing since "readonly folders" are broken since Windows 7 i think?
Basically, all folders are "read only" what changes nothing. And if you try to remove them this attribute it will be restored instantly by itself... Yeah, Microsoft...

**Review by kv2019i — APPROVED** [review]
@aborisovich please check

**Review by lgirdwood — APPROVED** [review]
